platform/net/ip/fakes/fake_ip_resolver.go

```diff
@@ -1,16 +1,20 @@
 package fakes
 
 import (
+	"fmt"
 	gonet "net"
 )
 
 type FakeResolver struct {
-	GetPrimaryIPv4InterfaceName string
-	GetPrimaryIPv4IPNet         *gonet.IPNet
-	GetPrimaryIPv4Err           error
+	GetPrimaryIPInterfaceName string
+	GetPrimaryIPNet           *gonet.IPNet
+	GetPrimaryIPErr           error
+	GetPrimaryIPCalledWith    []string
 }
 
-func (r *FakeResolver) GetPrimaryIPv4(interfaceName string) (*gonet.IPNet, error) {
-	r.GetPrimaryIPv4InterfaceName = interfaceName
-	return r.GetPrimaryIPv4IPNet, r.GetPrimaryIPv4Err
+func (r *FakeResolver) GetPrimaryIP(interfaceName string, is_ipv6 bool) (*gonet.IPNet, error) {
```

Comment:
can we choose `ipv6`, `isIpv6` (or perhaps `isIPv6`)... 

`is_ipv6` seems wrong in that Golang suggest camelCase for variables, as does `IsVersion6` (see `func (ia failingInterfaceAddress) GetIP(IsVersion6 bool) (string, error)`)